--- libXaw-1.0.10/src/DisplayList.c.bero	2012-03-29 20:48:13.080429703 +0200
+++ libXaw-1.0.10/src/DisplayList.c	2012-03-29 20:48:25.887430654 +0200
@@ -290,7 +290,7 @@ _XawDisplayList *XawCreateDisplayList(St
 	}
       if (fp)
 	{
-	  snprintf(cname, fp - fname + 1, fname);
+	  snprintf(cname, fp - fname + 1, "%s", fname);
 	  memmove(fname, fp + 1, strlen(fp));
 	  lc = cname[0] ? XawGetDisplayListClass(cname) : xlibc;
 	  if (!lc)
